init.c

```diff
@@ -955,18 +948,17 @@ int luaopen_libcutorch(lua_State *L)
   lua_setglobal(L, "cutorch");
   luaL_setfuncs(L, cutorch_stuff__, 0);
 
-  THCState* state = (THCState*)malloc(sizeof(THCState));
-  memset(state, 0, sizeof(THCState));
+  THCState* state = THCState_alloc();
```

Comment:
It's `alloc` because there's a separate initialization function. The `_new` functions initialize their data structure.